UnityProject/Assets/Scripts/Weapons/Projectiles/Behaviours/ProjectileDamageAtPoint.cs

```diff
@@ -0,0 +1,54 @@
+﻿using UnityEngine;
+
+namespace Weapons.Projectiles.Behaviours
+{
+	/// <summary>
+	/// Creates a raycast at a certain point and process IOnHit behaviours
+	/// Script is used by Proto-Kinetic-Accelerator for damaging everything on one tile
+	/// </summary>
+	public class ProjectileDamageAtPoint : MonoBehaviour, IOnShoot, IDespawn
+	{
+		private IOnHit[] behavioursOnBulletHit;
+		private Vector2 direction;
+
+		[SerializeField] private LayerMask layersToHit;
+
+		private void Awake()
+		{
+			behavioursOnBulletHit = GetComponents<IOnHit>();
+		}
+
+		public void OnShoot(Vector2 direction, GameObject shooter, Gun weapon, BodyPartType targetZone = BodyPartType.Chest)
+		{
+			this.direction = direction;
+		}
+
+		public void Despawn(RaycastHit2D hit, Vector2 point)
+		{
+			if (hit.collider == null)
+			{
+				RayCastAt(point);
+			}
+			else
+			{
+				var bulletHitTarget = Vector3.zero;
+				bulletHitTarget.x = hit.point.x - 0.01f * hit.normal.x;
+				bulletHitTarget.y = hit.point.y - 0.01f * hit.normal.y;
+				RayCastAt(bulletHitTarget);
+			}
+		}
+
+		private void RayCastAt(Vector2 point)
+		{
+			var rayHits = Physics2D.RaycastAll(point, direction,0f, layersToHit);
```

Comment:
mmm, certainly wouldn't trust raycasts `for damaging everything on one tile`, you'd be better off using MatrixManager.GetAt<>(), or matrix.Get<> if you know a particular matrix

Reply:
I was trying to start implementing logic for shooting like in old ss13. When you click on an object for example. If bullet travels to that point in the world where you clicked it, bullet will damage first object which is on this exact point.

But i realized that i will have to touch firmly `Gun.cs` which i did not want to dig myself into.

Reply:
so what's your plan on this?

Reply:
Leave it as it is. I will have to add another script with `MatrixManager.GetAt<>()` as you suggested. Or maybe your suggestion solves all problems and should be one script anyways.

The logic of clicking and shooting the point in world space where you clicked on, is supposed to get first object on the tile you clicked on.